src/extensions/canonicalResourceDirectiveExtension.ts

```diff
@@ -6,21 +6,20 @@ export const canonicalResourceDirectiveExtension = (documentAST, result) => {
   const canonicalFieldPath = getCanonicalResourceDirectiveForField(documentAST)
   let canonicalExtensions = {}
   if (canonicalFieldPath.length && result.errors && result.errors.length) {
-    result.errors.forEach(e => {
-      if (isEqual(e.path, canonicalFieldPath)) {
-        const errors = flattenErrors(e)
+    const errors = result.errors
+      .filter(e => isEqual(e.path, canonicalFieldPath))
+      .pop()
```

Comment:
Use `Array.prototype.find()` here, which, like `some`, will stop as soon as a match is found.